bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityTypeArithmeticGroupFunction.java

```diff
@@ -62,11 +65,26 @@ public State[] getParameters() {
             return new State[0];
         }
 
-        protected boolean isSameDimension(@Nullable Item item) {
-            if (item instanceof GroupItem groupItem) {
-                return isSameDimension(groupItem.getBaseItem());
+        protected boolean isCompatible(@Nullable Item member) {
+            if (member instanceof GroupItem groupItem) {
+                return isCompatible(groupItem.getBaseItem());
+            }
+            if ((baseItem instanceof NumberItem baseNI) && (member instanceof NumberItem memberNI)) {
+                Unit<? extends Quantity<?>> baseUnit = baseNI.getUnit();
+                Unit<? extends Quantity<?>> memberUnit = memberNI.getUnit();
+                if (baseUnit != null && memberUnit != null) {
+                    return baseUnit.isCompatible(memberUnit) || baseUnit.isCompatible(memberUnit.inverse());
```

Comment:
```suggestion
                    return baseUnit.isCompatible(memberUnit) || baseUnit.isCompatible(memberUnit.getBaseUnit().inverse());
```

We discussed this previously. I think we should always use the base unit for inversion, which works in all cases we have identified so far.

Reply:
I guess you meant `getSystemUnit()` here?

Reply:
PS Insofar as all the JUnit tests succeeded I am wondering if the change is necessary? Or should we add another test for that case?

Reply:
> I guess you meant `getSystemUnit()` here?

Yes, that's what I meant.


> PS Insofar as all the JUnit tests succeeded I am wondering if the change is necessary? Or should we add another test for that case?

Maybe try checking if `mK` (milliKelvin) `toInvertibleUnit` and then `toUnit(Mired)` works. I suspect that may give strange results as the inverse of milliKelvin is not in the same scale as Mired.

Reply:
milliKelvin is wrong for two reasons .. a) the factor would be micro rather than milli, and b) the conversions are `mirek = 1e6 / kelvin` resp. `kelvin = 1e6 / mired` -- i.e. not mirek = `microKelvin` ..

Reply:
My prior JUnit tests were testing `toInvertibleUnit` with target Kelvin and they all pass. However I added a test that tests `toInvertibleUnit` with target Mirek `toInvertibleUnit` with the inverse target Mirek. This test currently fails. However it should work fine once #4561 is merged.

Reply:
Maybe also keep in mind that `isCompatible()` works both ways even without falling back to `systemUnit`. The issue was that 'toInvertibleUnit` itself had a bug that requires an intermediate conversion via `systemUnit`.

So there is nothing to fix at this point in the code.

And #4561 provides the intermediate conversion via `systemUnit`.

Reply:
My intention was to not start from the system unit (K), but from a derived unit (like mK or whatever multiplier symbol). But you are right with the fix in toInvertibleUnit, it should be fine.

Reply:
> My prior JUnit tests were testing `toInvertibleUnit` with target Kelvin and they all pass. However I added a test that tests `toInvertibleUnit` with the inverse target Mirek. This test currently fails. However it should work fine once #4561 is merged.

I don’t get why that fix would make a difference though. Did you try running the test with the patch applied as well?

Reply:
Ok I will patch and test.

Reply:
I just tried the tests with the #4561 code patch applied, and I can confirm that it works fine.